lib/utils/saveProgress.ts

```diff
@@ -0,0 +1,78 @@
+import { type Language } from "@lib/types/form-builder-types";
+import { FormProperties } from "@cdssnc/gcforms-types";
+import { type FormValues } from "@lib/formContext";
+import { toggleSavedValues } from "@i18n/toggleSavedValues";
+
+export const SESSION_STORAGE_KEY = "form-data";
+
+export const removeProgressStorage = () => {
+  sessionStorage.removeItem(SESSION_STORAGE_KEY);
+};
+
+type Options = {
+  id: string;
+  values: FormValues;
+  history: string[];
+  currentGroup: string;
+};
+
+export const saveProgress = (
+  language: string = "en",
+  { id, values, history, currentGroup }: Options
+) => {
+  if (typeof sessionStorage === "undefined") {
+    return false;
+  }
+
```

Comment:
Suggestion for a future iteration. I wonder if we want any error catching here? I guess in the case of an error we'd just show an empty form. Or perhaps an error + empty form would be more helpful.